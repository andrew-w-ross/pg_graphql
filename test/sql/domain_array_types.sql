begin;

create schema other;

comment on schema public is e'@graphql({"inflect_names": true, "resolve_base_type": true})';

create domain int_array as int[];

create domain domain_int as int;

create domain domain_int_array as domain_int[];

create table domain_test (
  id serial primary key,
  field_int_array int_array,
  field_domain_int_array domain_int_array
);

select jsonb_pretty(
    graphql.resolve(
    $$
    {
      __type(name: "DomainTest") {
        fields {
          name
          type {
            kind
            ofType {
              kind
              name
              ofType {
                kind
                name
              }
            }
          }
        }
      }
    }
    $$
  )
);

rollback;