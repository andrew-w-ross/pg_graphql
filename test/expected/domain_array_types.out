begin;
create schema other;
comment on schema public is e'@graphql({"inflect_names": true, "resolve_base_type": true})';
create domain int_array as int[];
create domain domain_int as int;
create domain domain_int_array as domain_int[];
create table domain_test (
  id serial primary key,
  field_int_array int_array,
  field_domain_int_array domain_int_array
);
select jsonb_pretty(
    graphql.resolve(
    $$
    {
      __type(name: "DomainTest") {
        fields {
          name
          type {
            kind
            ofType {
              kind
              name
              ofType {
                kind
                name
              }
            }
          }
        }
      }
    }
    $$
  )
);
                    jsonb_pretty                    
----------------------------------------------------
 {                                                 +
     "data": {                                     +
         "__type": {                               +
             "fields": [                           +
                 {                                 +
                     "name": "nodeId",             +
                     "type": {                     +
                         "kind": "NON_NULL",       +
                         "ofType": {               +
                             "kind": "SCALAR",     +
                             "name": "ID",         +
                             "ofType": null        +
                         }                         +
                     }                             +
                 },                                +
                 {                                 +
                     "name": "id",                 +
                     "type": {                     +
                         "kind": "NON_NULL",       +
                         "ofType": {               +
                             "kind": "SCALAR",     +
                             "name": "Int",        +
                             "ofType": null        +
                         }                         +
                     }                             +
                 },                                +
                 {                                 +
                     "name": "fieldIntArray",      +
                     "type": {                     +
                         "kind": "LIST",           +
                         "ofType": {               +
                             "kind": "SCALAR",     +
                             "name": "Int",        +
                             "ofType": null        +
                         }                         +
                     }                             +
                 },                                +
                 {                                 +
                     "name": "fieldDomainIntArray",+
                     "type": {                     +
                         "kind": "LIST",           +
                         "ofType": {               +
                             "kind": "SCALAR",     +
                             "name": "Int",        +
                             "ofType": null        +
                         }                         +
                     }                             +
                 }                                 +
             ]                                     +
         }                                         +
     }                                             +
 }
(1 row)

rollback;
